comp/core/workloadmeta/collectors/internal/kubelet/kubelet.go

```diff
@@ -145,6 +145,12 @@ func (c *collector) parsePods(pods []*kubelet.Pod) []workloadmeta.CollectorEvent
 			&podID,
 		)
 
+		// Attempt to find the GPU type from running containers first, then init containers
+		GPUType := findPodGPUType(pod.Spec.Containers)
```

Comment:
since previously the container and initcontainer has been already parsed, maybe your function can use the `podInitContainers` and `podContainers` instead of using the raw data.

Reply:
Unfortunately those objects don't contain the GPU info in an accessible manner. They are of type:
```
// OrchestratorContainer is a reference to a Container with
// orchestrator-specific data attached to it.
type OrchestratorContainer struct {
	ID    string
	Name  string
	Image ContainerImage
}
```
I could add a GPU_Vendor string slice to this struct but this struct seems like its meant to be very simple.

Reply:
Ok make sense, 
I looked at the implementation of `parsePodContainers` and in fact the 'workloadmeta.Container' resource that I was think of is part of the containerEvents.
you can find it creation here https://github.com/DataDog/datadog-agent/blob/fc8aad3bfe40880c5b706d9484e3914530d4f046/comp/core/workloadmeta/collectors/internal/kubelet/kubelet.go#L290C26-L290C35 

it would be nice that you update the `parsePodContainers` and the `workloadmeta.Container` struct to add the `GPUVendor` field like you did for `workloadmeta.Pod`

Reply:
The GPUVendor field is already inside of the `workloadmeta.Container` struct via the `Resources` field which is of type `struct ContainerResources`. It made sense to include it in the Resource struct since it's defined under Resources in the manifest.

The issue is that `workloadmeta.CollectorEvent` stores the `workloadmeta.Container` as a `workloadmeta.Entity`. I suppose I could type cast when I want to look for the field with something like `containerEvents[i].Entity.(*workloadmeta.Container).Resources.GPURequest
`